fault/tester.py

```diff
@@ -126,8 +127,11 @@ def make_target(self, target: str, **kwargs):
         raise NotImplementedError(target)
 
     def is_recursive_type(self, T):
+        if isinstance(T, fault.WrappedVerilogInternalPort):
+            return False
         return isinstance(T, m.TupleType) or isinstance(T, m.ArrayType) and \
-            not isinstance(T.T, (m._BitKind, m.BitType))
+            not isinstance(T.T, (m._BitKind, m.BitType)) or \
```

Comment:
This seems like a pretty nasty condition.   Is this something we could push into magma to simplify life?

Reply:
I feel like any time a type is added this will have to change.  Which is bad.

Reply:
I don't know if there's a way around it.  Fault needs to understand recursive types.  If a type subclasses the recursive types, that should work, but if a new type is added, fault has no way of knowing how to handle it unless we can define some generic traversal function.  I think the way to avoid the condition would be some way to generically visit a recursive type.

Reply:
> I think the way to avoid the condition would be some way to generically visit a recursive type.
 
In hwtypes adts define `field_dict` to make this type of recursion easier.   Maybe magma could have a similar sort of protocol?

Reply:
Sure, but I'd separate those changes out from this as this adds the basic functionality, while that involves a refactor of a lot of internal code to adhere to the new interface

Reply:
Of course.  This was more of an observation than a request for code changes.  I generally dislike complicated if conditions because they break and are generally a sign of poor code structuring.  Regardless, neither magma nor fault are mine; doesn't matter to me if *you* have to write a million ifs.